idl2library.cmake

```
@@ -0,0 +1,226 @@
+        list(APPEND            ALL_IDL_RELDIRS ${${current_idl_target}_RELDIR})
+        list(APPEND            ALL_IDL_PATHS ${${current_idl_target}_ABSPATH})
+    endforeach()
+    list(REMOVE_DUPLICATES ALL_IDL_RELDIRS)
+    list(REMOVE_DUPLICATES ALL_IDL_PATHS)
+
+    #Add a library to build for each input idl
+    foreach(SINGLE_IDL ${IDL_WISHLIST})
+        get_idl_target_name(${SINGLE_IDL} current_idl_target)
+
+        # If this target has already been created, we're done
+        list(FIND IDL_TARGETS ${current_idl_target} SEARCH_INDEX)
+        if(${SEARCH_INDEX} GREATER -1)
+            unset(${current_idl_target}_ABSPATH)
+            message(WARNING "Already created a target for ${current_idl_target}..  Skipping")
+            break()
+        endif(${SEARCH_INDEX} GREATER -1)
+
+        # Second duplicate target check
+        if(TARGET ${current_idl_target})
+            unset(${current_idl_target}_ABSPATH)
+            message(WARNING "Already created a target for ${current_idl_target}..  Skipping")
+            break()
+        endif(TARGET ${current_idl_target})
+
+	    unset(IDL_TARGET_DEPENDENCIES)
+	    unset(current_idl_include_opts)
+
+        find_idl_dependencies("${${current_idl_target}_ABSPATH}") #Dependencies are returned in IDL_TARGET_DEPENDENCIES
+		
+		#Note:  current_idl_include_opts needs to be a list.  Previously it was a string and that will no longer work correctly.
+        foreach(target_dependency ${IDL_TARGET_DEPENDENCIES})
+            if(NOT "${${target_dependency}_ABSDIR}" STREQUAL "${${current_idl_target}_ABSDIR}")
+				list(APPEND current_idl_include_opts "-I${${target_dependency}_ABSDIR}")
+            endif()
+        endforeach()
+		list(REMOVE_DUPLICATES current_idl_include_opts)
+		
+		if(OPENDDS_CPP11_IDL_MAPPING)
+            list(APPEND current_idl_include_opts "-Lc++11 ")
+        endif()
+		
+        message("Adding library: ${current_idl_target}")
+		message("current_idl_include_opts:  ${current_idl_include_opts}")
+        message("Dependencies: ${IDL_TARGET_DEPENDENCIES}\n")
+
+        if(idl2library_BUILD_STATIC)
+            add_library(${current_idl_target} STATIC)
+			set_property(TARGET ${current_idl_target} PROPERTY POSITION_INDEPENDENT_CODE ON)
+		else()
+	        add_library(${current_idl_target} SHARED)
+		endif()
+
+        OPENDDS_TARGET_SOURCES(${current_idl_target} 
+            ${${current_idl_target}_RELPATH}
+            OPENDDS_IDL_OPTIONS ${current_idl_include_opts}
+			TAO_IDL_OPTIONS ${current_idl_include_opts}
+        )
+        target_link_libraries(${current_idl_target}
+           ${IDL_TARGET_DEPENDENCIES}
+            OpenDDS::Dcps
+        )
+
+        # Group the IDL projects together
+        set_target_properties(${current_idl_target} PROPERTIES FOLDER IDL)
+        target_compile_definitions(${current_idl_target} PUBLIC __OPENDDS_IDL_HAS_ANNOTATIONS _HAS_AUTO_PTR_ETC=1 _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
```

```suggestion
        target_compile_definitions(${current_idl_target} PUBLIC  _HAS_AUTO_PTR_ETC=1 _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
```

I'm not sure what this is supposed to do. `__OPENDDS_IDL_HAS_ANNOTATIONS` is for IDL and this function is for C++ macro definitions.

---

Removed.